File: ServicesApp/app/src/main/java/com/example/servicesapp/MainActivity.java
package com.example.servicesapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btnStart = findViewById(R.id.btn_start);
        btnStart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent serviceIntent = new Intent(MainActivity.this, PlayerService.class);
                startService(serviceIntent);
            }
        });

        Button btnStop = findViewById(R.id.btn_stop);
        btnStop.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent serviceIntent = new Intent(MainActivity.this, PlayerService.class);
                stopService(serviceIntent);
            }
        });

    }
}

File: MealApp/app/src/main/java/com/example/mealapp/MainActivity.kt
package com.example.mealapp

import android.os.Bundle
import android.view.Menu
import android.view.View
import com.example.mealapp.ui.base.BaseActivity
import com.example.mealapp.ui.base.BaseFragment
import com.example.mealapp.ui.explore.ExploreFragment
import com.example.mealapp.ui.favorites.FavoritesFragment
import com.example.mealapp.ui.home.HomeFragment
import com.example.mealapp.ui.more.MoreFragment
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        bottom_bar.setOnNavigationItemSelectedListener {
            when (it.itemId) {
                R.id.home -> replaceFragment(HomeFragment(), R.id.home_container)
                R.id.explore -> replaceFragment(ExploreFragment(), R.id.explore_container)
                R.id.favorites -> replaceFragment(FavoritesFragment(), R.id.favorites_container)
                R.id.more -> replaceFragment(MoreFragment(), R.id.more_container)
                else -> false
            }
        }
        bottom_bar.selectedItemId = R.id.home
    }

    private fun replaceFragment(fragment: BaseFragment, container: Int): Boolean {
        var fragmentToReplace = fragment
        var topFragment = supportFragmentManager.findFragmentById(container)
        if (topFragment is BaseFragment) {
            fragmentToReplace = topFragment
        }

        supportFragmentManager.beginTransaction()
            .replace(container, fragmentToReplace)
            .commit()

        favorites_container.visibility = View.GONE
        explore_container.visibility = View.GONE
        home_container.visibility = View.GONE
        more_container.visibility = View.GONE
        findViewById<View>(container).visibility = View.VISIBLE

        return true
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.toolbar_menu, menu)
        return true
    }
}
